api/client/proto/authservice.proto

```diff
@@ -897,12 +897,24 @@ message DeleteMFADeviceSyncRequest {
     // TokenID is the ID of a user token that will be used to verify this request.
     // Token types accepted are:
     //   - Recovery approved token that is obtained with RPC ApproveAccountRecovery
-    //   - TODO(kimlisa) add privilege token doc after implementation
+    //   - Privilege token that is obtained with RPC CreatePrivilegeToken
     string TokenID = 1 [ (gogoproto.jsontag) = "token_id" ];
     // DeviceName is the name of the device to delete.
     string DeviceName = 2 [ (gogoproto.jsontag) = "device_name" ];
 }
 
+// AddMFADeviceSyncRequest is a request to add a MFA device (nonstream).
+message AddMFADeviceSyncRequest {
+    // PrivilegeTokenID is the ID of a privilege token that will be used to verify this request.
+    // Privilege tokens are obtained after a logged in user re-authenticates with their second
+    // factor.
+    string PrivilegeTokenID = 1 [ (gogoproto.jsontag) = "privilege_token_id" ];
+    // NewDeviceName is the name of a new mfa device.
+    string NewDeviceName = 2 [ (gogoproto.jsontag) = "new_device_name,omitempty" ];
+    // NewMFAResponse is a user's new mfa response to a mfa register challenge.
+    MFARegisterResponse NewMFAResponse = 3 [ (gogoproto.jsontag) = "new_mfa_response,omitempty" ];
```

Comment:
for u2f, front will call `h.GET("/webapi/u2f/signuptokens/:token", httplib.MakeHandler(h.u2fRegisterRequest))` which will call in back [CreateSignupU2FRegisterRequest](https://github.com/gravitational/teleport/blob/master/lib/auth/password.go#L353)

for totp, front will call `h.POST("/webapi/mfa/token/:token/totpsecrets", h.WithAuth(h.createTOTPSecretsHandle))`
which will call in back [RotateUserTokenSecrets](https://github.com/gravitational/teleport/blob/master/lib/auth/usertoken.go#L233)

Reply:
I would like to unify all of those, plus webauthn, in a single endpoint/RPC (`/webapi/mfa/token/:token/registerchallenge` / `CreateRegisterChallenge`). Let's talk about that, it would be great if you could knock that out for us during this work.

Reply:
implemented here: https://github.com/gravitational/teleport/pull/8342